orttraining/orttraining/python/training/orttrainer_options.py

```diff
@@ -425,12 +491,63 @@ def _check_is_callable(field, value, error):
                 'min': 0,
                 'default': 0
             },
+            'data_parallel_size': {
+                'type': 'integer',
+                'min': 1,
+                'default': 1
+            },
+            'horizontal_parallel_size': {
+                'type': 'integer',
+                'min': 1,
+                'default': 1
+            },
+            'pipeline_parallel_size': {
```

Comment:
You have to either define `pipiline_parallel_size` and `pipeline_parallel` dependency on the schema, expliciting that when `pipeline_parallel_size > 1` `pipeline_parallel` cant be empty.

Another simpler idea that at mitigates the risk of user setting `pipeline_parallel_size > 1` and forggeting about `pipeline_parallel` is to move `pipeline_piralel_size` inside `pipeline_parallel`